Automattic/wp-calypso
# Gutenboarding: Use long mshots previews

#### Changes proposed in this Pull Request

This PR utilises longer screenshots made available from updates to the mShots service.

#### Testing instructions

The happy path here is to load up `/new/design?flags=gutenboarding/long-previews` and check that the hover animations look good.

Note that the `/` in the feature flags gets encoded as the calypso.live links are resolved, so you'll need to fix those in test urls.

In addition, there are quite a few variations to check that we don't break:
- [ ] `&flags=gutenboarding/site-editor`: The wip-FSE view has only a few different featured designs
- [ ] `&anchor_podcast=1234567`: The Anchor onboarding design grid has a different set of designs and limits the number of columns
- [ ] Navigate to some other locales, e.g.
 - http://calypso.localhost:3000/new/design/ar?flags=gutenboarding/long-previews
 - http://calypso.localhost:3000/new/design/zh-tw?flags=gutenboarding/long-previews
 - http://calypso.localhost:3000/new/design/es?flags=gutenboarding/long-previews
- [ ] Check the static vs the dynamic previews. There are three designs `available-designs-config.json` with the `"preview": "static"` flag that disables the animations and also need special handling during loading: Pollard, Fredrickson and Balan.
- [ ] Check various screen-sizes, particularly the ones with max/min frame width just before/after the number of columns switches.
- [ ] Check the loading state. This PR includes a `gutenboarding/bust-mshots-cache` feature flag to help with testing here, e.g. http://calypso.localhost:3000/new/design?flags=gutenboarding/long-previews,gutenboarding/bust-mshots-cache . Another good trick is to block the urls, as this will keep them in the loading state forever. You can block specific urls or you can edit the domain to block all of them:
 
![WordPress_com_and_wp-calypso_—_node_◂_node__usr_local_Cellar_yarn_1_22_5_libexec_bin_yarn_js_start_—_127×33_and_WordPress_com](https://user-images.githubusercontent.com/5952255/114190386-98a09a80-998e-11eb-9e0d-d66baffc269d.jpg)

The new functionality is behind the `gutenboarding/long-previews` flag. In addition, there are two other config flags introduced in this PR to help with testing:
`gutenboarding/gutenboarding/bust-mshots-cache`: Generates a new unique mshots url for each page-load to ease testing the uncached images functionality.
`gutenboarding/local-mshots`: Redirects mshots requests to a local dev server (see 31-gh-Automattic/mShots)

#### TODO:
- [x] Constrain changes to the feature flag (currently there are a number of css and className changes in effect without the flag)
- [x] Check that all the screenshots look good. They look great in landscape, portrait not so much. I think going back to the portraits needs another look from some designers after this PR lands.
 - [x] See if there's an easy fix for extra blank length on Pollard and Balan. Same note as above: These `static` themes are the worst offenders with portrait mode.

- [x] ~Restore fading in/out transitions when screenshots are loaded~
This is proving pretty tricky. After rebasing this PR 3 times now, I'm wondering if it would be better to do this in a follow-up PR: Punted to https://github.com/Automattic/wp-calypso/pull/51918
